zaza/openstack/charm_tests/ceph/fs/tests.py

```diff
@@ -103,6 +107,64 @@ def verify(stdin, stdout, stderr):
             'sudo cat /mnt/cephfs/test',
             password=password, privkey=privkey, verify=verify)
 
+    def test_conf(self):
+        """Test ceph to ensure juju config options are properly set."""
+        self.TESTED_UNIT = 'ceph-fs/0'
+
+        def _get_conf():
+            """get/parse config file into dict for specified configs.
+
+            :returns dict: conf options selected from configs
+            :rtype: dict
+            """
+            conf = model.run_on_unit(self.TESTED_UNIT, "cat {}"
+                                         .format('/etc/ceph/ceph.conf'))
+            holder = []
+            configs = ["mds cache memory limit",
+                       "mds cache reservation",
+                       "mds health cache threshold"]
+            for item in conf['Stdout'].split("\n"):
+                if any(ext in item for ext in configs):
+                    holder.append(tuple(item.split('=')))
+            # strip the leading/trailing whitespace
+            return dict((k.strip(), v.strip())
+                        for k, v in dict(holder).items())
+
+        def _change_conf_check(mds_config):
+            """Change configs, then assert to ensure config was set.
+
+            Doesn't return a value.
+            """
+            loop = asyncio.get_event_loop()
+            crt = model.async_set_application_config('ceph-fs', mds_config)
+            loop.run_until_complete(crt)
+            results = _get_conf()
+            time.sleep(1)  # needs time to release lock and update conf.
```

Comment:
Using sleeps like this to assert values later can be rather fragile. I suggest using `tenacity` and retrying a couple of times to make the test more robust.